pages/spaceship.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useState, useEffect } from 'react';
+import Layout from '../app/layout';
+import SelectLayout from "@/app/selectLayout";
+import ActionButton from "@/app/components/actionButton/ActionButton";
+import SelectItems from "@/app/components/selectItems/selectItems";
+import "../app/globals.css";
+import Image from "next/image";
+
+interface Color {
+    code: string;
+    word: string;
+}
+
+const Spaceship: React.FC = () => {
+    const [selectedImageIndex, setSelectedImageIndex] = useState<number>(0);
+    const [selectedColorIndex, setSelectedColorIndex] = useState<number>(0);
+    const [isClient, setIsClient] = useState<boolean>(false);
+
+    useEffect(() => {
+        setIsClient(true);
+
+        const storedImageIndex = localStorage.getItem('spaceshipSelectedImageIndex');
+        if (storedImageIndex) {
+            setSelectedImageIndex(parseInt(storedImageIndex));
+        }
+
+        const storedColorIndex = localStorage.getItem('spaceshipSelectedColorIndex');
+        if (storedColorIndex) {
+            setSelectedColorIndex(parseInt(storedColorIndex));
+        }
+    }, []);
+
+    const handleImageClick = (index: number) => {
+        setSelectedImageIndex(index);
+        localStorage.setItem('spaceshipSelectedImageIndex', index.toString());
+    };
+
+    const handleColorClick = (index: number) => {
+        setSelectedColorIndex(index);
+        localStorage.setItem('spaceshipSelectedColorIndex', index.toString());
+    };
+
+    const colors: Color[] = [
+        { code: "#FF9900", word: "orange" },
+        { code: "#5EC71E", word: "green" },
+        { code: "#0E5CF3", word: "blue" },
+        { code: "#CA0000", word: "red" },
+        { code: "#FF00F5", word: "pink" },
+        { code: "#7A41F3", word: "purple" },
+        { code: "#FFE500", word: "yellow" }
+    ];
+
+    const renderLeftChildren = () => {
+        if (isClient) {
+            const selectedImage = localStorage.getItem('spaceshipSelectedImageIndex') ? parseInt(localStorage.getItem('spaceshipSelectedImageIndex')!) : selectedImageIndex;
+            const selectedColor = localStorage.getItem('spaceshipSelectedColorIndex') ? parseInt(localStorage.getItem('spaceshipSelectedColorIndex')!) : selectedColorIndex;
+            return (
+                <div>
+                    <div className="text-center mt-2">
```

Comment:
Adapt Rocket positioning its not in the middle maybe use h-full w-auto flex justify-center items-center